src/sqlfluff/dialects/dialect_mysql.py

```diff
@@ -310,7 +310,7 @@ class TableConstraintSegment(BaseSegment):
         Sequence(  # [ CONSTRAINT <Constraint name> ]
             "CONSTRAINT", Ref("ObjectReferenceSegment"), optional=True
         ),
-        OneOf(
+        Delimited(
```

Comment:
Why was this change needed? (I.e. what test cases / syntax?) From the original issue description, I expected the changes below (line 338...), but not this one.

Reply:
Pulled in the examples for this link: https://dev.mysql.com/doc/refman/8.0/en/create-table-foreign-keys.html

It shows comma delimited values are allowed for PRIMARY KEY/INDEX/FOREIGN KEY.

For example (this has been added as a test case in this PR btw):

```sql
CREATE TABLE product_order (
    no INT NOT NULL AUTO_INCREMENT,
    product_category INT NOT NULL,
    product_id INT NOT NULL,
    customer_id INT NOT NULL,

    PRIMARY KEY(no),
    INDEX (product_category, product_id),
    INDEX (customer_id),

    FOREIGN KEY (product_category, product_id)
      REFERENCES product(category, id)
      ON UPDATE CASCADE ON DELETE RESTRICT,

    FOREIGN KEY (customer_id)
      REFERENCES customer(id)
)
```

Reply:
Got it -- thanks!